lightning/src/ln/msgs.rs

```diff
@@ -1194,6 +1432,130 @@ impl_writeable_msg!(AcceptChannel, {
 	(1, channel_type, option),
 });
 
+impl_writeable_msg!(AcceptChannelV2, {
+	temporary_channel_id,
+	funding_satoshis,
+	dust_limit_satoshis,
+	max_htlc_value_in_flight_msat,
+	htlc_minimum_msat,
+	minimum_depth,
+	to_self_delay,
+	max_accepted_htlcs,
+	funding_pubkey,
+	revocation_basepoint,
+	payment_basepoint,
+	delayed_payment_basepoint,
+	htlc_basepoint,
+	first_per_commitment_point,
+	shutdown_scriptpubkey
+}, {
+	(1, channel_type, option),
+});
+
+impl Writeable for TxAddInput {
+	fn write<W: Writer>(&self, w: &mut W) -> Result<(), io::Error> {
+		self.channel_id.write(w)?;
+		self.serial_id.write(w)?;
+		(self.prevtx.serialized_length() as u16).write(w)?;
+		self.prevtx.write(w)?;
+		self.prevtx_out.write(w)?;
+		self.sequence.write(w)?;
+		Ok(())
+	}
+}
+
+impl Readable for TxAddInput {
+	fn read<R: Read>(r: &mut R) -> Result<Self, DecodeError> {
+		let channel_id = Readable::read(r)?;
+		let serial_id = Readable::read(r)?;
+		let prevtx_len: u16 = Readable::read(r)?;
+		let mut prevtx_bytes: Vec<u8> = Vec::with_capacity(prevtx_len as usize);
+		for _ in 0..prevtx_len {
+			prevtx_bytes.push(Readable::read(r)?);
+		}
+		let prevtx = deserialize(&prevtx_bytes).map_err(|_| DecodeError::InvalidValue)?;
+		let prevtx_out = Readable::read(r)?;
+		let sequence = Readable::read(r)?;
+
+		Ok(Self {
+			channel_id, serial_id, prevtx, prevtx_out, sequence
+		})
+	}
+}
+
+impl_writeable_msg!(TxAddOutput, {
+	channel_id,
+	serial_id,
+	sats,
+	script,
+}, {});
+
+impl_writeable_msg!(TxRemoveInput, {
+	channel_id,
+	serial_id,
+}, {});
+
+impl_writeable_msg!(TxRemoveOutput, {
+	channel_id,
+	serial_id,
+}, {});
+
+impl_writeable_msg!(TxComplete, {
+	channel_id,
+}, {});
+
+impl_writeable_msg!(TxSignatures, {
+	channel_id,
+	tx_hash,
+	witnesses,
+}, {});
+
+impl Writeable for Vec<Witness> {
+	fn write<W: Writer>(&self, w: &mut W) -> Result<(), io::Error> {
+		(self.len() as u16).write(w)?;
+		for witness in self {
+			(witness.len() as u16).write(w)?;
+			for element in witness.iter() {
+				element.to_vec().write(w)?;
+			}
+		}
+		Ok(())
+	}
+}
+
+impl Readable for Vec<Witness> {
+	fn read<R: Read>(r: &mut R) -> Result<Self, DecodeError> {
+		let num_witnesses: u16 = Readable::read(r)?;
+		let mut witnesses = Vec::with_capacity(num_witnesses as usize);
+		for _ in 0..num_witnesses {
+			let num_elements: u16 = Readable::read(r)?;
```

Comment:
> Can we convince upstream to use the bitcoin consensus serialization for witnesses?

I will try :)

Reply:
Hmm, so how would that look in the spec? Is it ok to state that the witness is a bunch of bytes with consensus encoding? Since we don't have `CompactSize` natively it seems like it's pretty much all we can do.

Reply:
Ok I think that's the best way to go to remain explicit. Will see what others think.

Reply:
So it appears that eclair and cln aren't currently exposed to consensus serialization at the moment and there is a question about consistency with other LN types. Although if I understand correctly, `prevtx` in `tx_add_input` _is_ consensus serialized (with witnesses)... Just says "serialized transaction" in the spec.

Reply:
So it [appears](https://github.com/lightning/bolts/pull/851/files#r1095521991) that eclair and cln aren't exposed to consensus serialization and there's a question about consistency with other LN types... But as far as I can see, `prevtx` in `tx_add_input` is in fact consensus serialized (with witnesses).